src/DotVVM.Framework/Controls/Infrastructure/HeadResourceLinks.cs

```diff
@@ -15,15 +15,22 @@ public class HeadResourceLinks : DotvvmControl
     {
         protected override void RenderControl(IHtmlWriter writer, IDotvvmRequestContext context)
         {
-            // render resource links
             var resourceManager = context.ResourceManager;
             if (resourceManager.HeadRendered) return;
-            resourceManager.HeadRendered = true; // set the flag before the resources are rendered, so they can't add more resources to the list during the render
+            // set the flag before the resources are rendered, so they can't add more resources to the list during the render
+            resourceManager.HeadRendered = true;
+
+            // render resource links and preloads
             foreach (var resource in resourceManager.GetNamedResourcesInOrder())
             {
                 if (resource.Resource.RenderPosition == ResourceRenderPosition.Head)
                     resource.RenderResourceCached(writer, context);
+
+                if (resource.Resource is IPreloadResource preloadResource)
```

Comment:
I think there should be a `else if`, we don't want to "prefetch" resources that are rendered above in the header.